hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java

```diff
@@ -125,15 +135,21 @@ public ResponseEntity<List<MgmtArtifact>> getArtifacts(
     // subroutine
     @SuppressWarnings("squid:S3655")
     public ResponseEntity<MgmtArtifact> getArtifact(@PathVariable("softwareModuleId") final Long softwareModuleId,
-            @PathVariable("artifactId") final Long artifactId) {
+            @PathVariable("artifactId") final Long artifactId,
+            @RequestParam(value = MgmtRestConstants.ARTIFACT_DOWNLOAD_URL_TYPE, required = false) final String artifactDownloadUrlType) {
         final SoftwareModule module = findSoftwareModuleWithExceptionIfNotFound(softwareModuleId, artifactId);
 
-        final MgmtArtifact reponse = MgmtSoftwareModuleMapper.toResponse(module.getArtifact(artifactId).get());
+        final MgmtArtifact response = MgmtSoftwareModuleMapper.toResponse(module.getArtifact(artifactId).get());
         if (!module.isDeleted()) {
-            MgmtSoftwareModuleMapper.addLinks(module.getArtifact(artifactId).get(), reponse);
+            if(artifactDownloadUrlType == null || artifactDownloadUrlType == "default") {
```

Comment:
No matter if we have "?downloadurltype=cdn" or "?downloadurltype=cdnnnnnnnnn", the result will be the same. Why we need  "?downloadurltype=default" option. Wouldn't be enough if we have default behaviour without parameter value(parameter is null)

Reply:
I agree, maybe it's not the best solution to have a String parameter here and we should use a boolean instead. Something like useCdn = true|false

Reply:
if we stick to strings, please use "mystring".equals(...)

Reply:
The difference is that, with the parameter being set to "cdn", the ArtifactUrlHandler is invoked internally to create the download URLs. So from a technical perspective, it would make sense to call the parameter "useArtifactUrlHandler=true|false". From an API usage perspective, however, sth with "CDN" might be more comprehensible...